NVK.cpp

```diff
@@ -1151,7 +1151,12 @@ bool NVK::utInitialize(WindowSurface* pWindowSurface)
         for(int i=0; i<device_extensions.size(); i++)
         {
             LOGI("%d: HW Device Extension: %s\n", i,device_extensions[i].extensionName);
-            device_extension_names[j][i] = device_extensions[i].extensionName;
+            // When device_extensions leaves the scope, VkExtensionProperties will free extensionName, so we need to copy it.
+            // We cannot free this data within this method, because that causes a crash after vkCreateDevice.
+            int len = strlen(device_extensions[i].extensionName) + 1; // +1 for \0
+            char* copiedName = new char[len];
+            std::memcpy(copiedName, device_extensions[i].extensionName, len);
+            device_extension_names[j][i] = copiedName;
```

Comment:
Memory leak occurs.
Maybe define `device_extension_names` as
```cpp
std::vector<std::vector<std::string>> device_extension_names;
```
could be simpler.

Reply:
I fixed the leak 😊.
Unfortunately, devInfo.ppEnabledExtensionNames is a char**, so we still need an array/vector of pointers (line 1193-1195).

In my previous test I got an error when using std::vector<char*>, because I assumed std::vector(length) sets the capacity to length, and the size to 0, but it sets both capacity and size to length 😅. This caused the first elements to be null, because I was using push_back().

Reply:
Yes, it is upset that `char**` is not compatible with `std::vector<std::string>`.